src/messaging/ExchangeContext.h

```diff
@@ -108,8 +111,8 @@ class DLL_EXPORT ExchangeContext : public ReferenceCounted<ExchangeContext, Exch
      *  @retval  #CHIP_NO_ERROR                             if the CHIP layer successfully sent the message down to the
      *                                                       network layer.
      */
-    CHIP_ERROR SendMessage(Protocols::Id protocolId, uint8_t msgType, System::PacketBufferHandle msgPayload,
-                           const SendFlags & sendFlags);
+    virtual CHIP_ERROR SendMessage(Protocols::Id protocolId, uint8_t msgType, System::PacketBufferHandle msgPayload,
```

Comment:
Why make ExchangeContext virtual? it seems this increase the size of ExchangeContext, I don't see any class is inherited from this class.

Reply:
This was needed to compile PASESession unit tests. I considered using `MessagingContext`. But that is currently inside src/messaging/test layer. Maybe we can move it out in a separate PR so that it can be used in other tests.

Reply:
Lets look into that,if a class is not designed to be inherited, why make it virtual, since we have 16 ExchangeContext by default, the size increase here does not make sense to me.

Reply:
Removed the virtual functions from this class for now. The PASE session unit test is disabled, with a TODO to enable it.